### PR TITLE
Upgrade: Go v1.20 and packages

### PR DESCRIPTION
**Pull request by dhaiducek**
ref: https://issues.redhat.com/browse/ACM-5042

### CODE REVIEW

File: client/source_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ type controllerRuntimeReconciler struct {
 	ReconcileCount int
 }
 
-func (r *controllerRuntimeReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+func (r *controllerRuntimeReconciler) Reconcile(_ context.Context, _ reconcile.Request) (reconcile.Result, error) {
```
Comment (yiraeChristineKim) [inline client/source_test.go:23]: why use _ instead of remove these

Comment (dhaiducek) [reply client/source_test.go:23]: I probably could have removed them, but I figured I'd maintain the footprint of the `Reconcile()` function to be clearer:
https://github.com/stolostron/kubernetes-dependency-watches/blob/3bcdbe44495e26bedb4fba027237041d1a506a6c/client/source.go#L39-L43

**Review by mprahl — APPROVED** [review]

**Pull request merged**